ansible/roles/agnosticd_save_output_dir/tasks/create-output-dir-archive.yml

```diff
@@ -18,4 +18,16 @@
     # Disable warning for using tar command rather than archive module.
     # archive module does not support chdir type option.
     warn: false
+
+- when: agnosticd_save_output_dir_archive_password is defined
+  block:
+    - name: Encrypt tarball using password
+      command: >-
+        gpg --symmetric --batch --passphrase-fd 0
+        --output {{ (agnosticd_save_output_dir_archive_tempfile ~ '.gpg') | quote }}
+        {{ agnosticd_save_output_dir_archive_tempfile | quote }}
+      args:
+        chdir: "{{ output_dir }}"
+        stdin: "{{ agnosticd_save_output_dir_archive_password }}"
+        creates: "{{ agnosticd_save_output_dir_archive_tempfile }}.gpg"
```

Comment:
Why a block with only one task?
Also, I don't think you want the command `args` `creates` here as that would prevent updating a `.gpg` file if somehow one happened to already exist.

Reply:
Good point for the block, it used to be another task that i deleted since, leaving only one. Fixed.

Reply:
for the `command.args.creates`,  i added them mainly because gnupg would fail if the file exist, and also because the files should always be absent anyway.

I removed them and added `--yes` to gpg commands to force overwrite.